cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/BullseyeParser.java

```diff
@@ -200,46 +195,96 @@ private void saveConditions(CoverageMeasures fileMeasuresBuilderIn) {
     totalcoveredconditions = 0;
   }
 
-  private void updateMeasures(String kind, String event, String line, CoverageMeasures fileMeasuresBuilderIn) {
+  private static void updateMeasures(String kind, String event, String line, CoverageMeasures fileMeasuresBuilderIn) {
 
-    if ("decision".equalsIgnoreCase(kind) || "condition".equalsIgnoreCase(kind)) {
-      if ("condition".equalsIgnoreCase(kind)) {
+    switch (kind.toLowerCase(Locale.ENGLISH)) {
+      case "catch":
+      case "decision":
+      case "for-range-body":
+      case "switch-label":
+      case "try":
+        totaldecisions++;
+        setTotalCoveredDecisions(event);
+        break;
+      case "condition":
         totalconditions += 2;
-        totalcoveredconditions += 1;
+        setTotalCoveredConditions(event);
+        break;
+      case "function":
         if ("full".equalsIgnoreCase(event)) {
-          totalcoveredconditions += 1;
-        }
-        if ("none".equalsIgnoreCase(event)) {
-          totalcoveredconditions -= 1;
+          fileMeasuresBuilderIn.setHits(Integer.parseInt(line), 1);
         }
-      } else {
-        totaldecisions += 1;
+        break;
+      case "constant":
+        break;
+    default:
+        LOG.warn("BullseyeParser unknown probe kind '{}'", kind);
+    }
+  }
+
+  /**
+   * @param event
+   */
+  private static void setTotalCoveredConditions(String event) {
+    switch (event.toLowerCase(Locale.ENGLISH)) {
+      case "full":
+        totalcoveredconditions += 2;
+        break;
+      case "true":
+      case "false":
+        totalcoveredconditions++;
+        break;
+      case "none":
+        // do nothing
+        break;
+      default:
+        // do nothing
```

Comment:
Add also a warning like for kind?

Reply:
The event list will not change (event has one of: none, false, true, full).

Reply:
Yes but default means also new/unknown kind?

Reply:
Yes, this is correct and typically a throw _IllegalArgumentException_ should be used. I use a silent ignore instead because many rules mandate the default switch clause (best practice) and a exception in BullseyeParser might not be a good idea.

Reply:
Still think there should be:
```Java
default:
   LOG.warn("BullseyeParser unknown probe event '{}'", event);
```